modules/rational-compile.el

```diff
@@ -63,16 +66,28 @@
 ;;;; Functions:
 ;; A function to locate a module's source file:
 (defun rational-locate-module-file (module &optional path)
-  (setq-local module-path nil)
-  (dolist (dir (or path
-                   (flatten-list `(,rational-compile-modules-path
-                                   ,rational-compile-extra-directories-list))))
-    (let ((file (expand-file-name (format "%s.el" (symbol-name module)) dir)))
-      (when (and (file-exists-p file)
-                 (not module-path))
-        (setq-local module-path file))))
-  module-path)
+  "Locates the source-file for the MODULE.
+
+MODULE must be a symbol.
+
+If the optional argument PATH is non-nil, it asumes it is a list
+containing the name of the directories to search for the
+source-file.  If it is ommited or nil, it will search in the
+directories defined in `rational-compile-modules-path' and
+`rational-compile-extra-directories-list'.
 
+It returns the first source-file that matches for the MODULE in
+the order specified search path.  If it finds none, it retorns
+nil."
+  (let ((file (expand-file-name
+               (format "%s.el" (symbol-name module))
+               (seq-find (lambda (dir)
+                           (file-exists-p (expand-file-name (format "%s.el" (symbol-name module)) dir)))
+                         (or path
+                             (flatten-list `(,rational-compile-modules-path
+                                             ,rational-compile-extra-directories-list)))))))
+      (when (file-exists-p file)
```

Comment:
The `when` here is probably not needed (duplicates the check in the `lambda`). `file` will either be a fully qualified file name, or it will be `nil` based on the lambda passed to `seq-find`.

Reply:
Yes! I didn't see that.